src/diffusers/models/resnet.py

```diff
@@ -673,6 +673,88 @@ def forward(self, x):
         return x
 
 
+class GlobalResponseNorm(nn.Module):
+    "Taken from https://github.com/facebookresearch/ConvNeXt-V2/blob/3608f67cc1dae164790c5d0aead7bf2d73d9719b/models/utils.py#L105"
+
+    def __init__(self, dim):
+        super().__init__()
+        self.gamma = nn.Parameter(torch.zeros(1, 1, 1, dim))
+        self.beta = nn.Parameter(torch.zeros(1, 1, 1, dim))
+
+    def forward(self, inputs):
+        Gx = torch.norm(inputs, p=2, dim=(1, 2), keepdim=True)
+        Nx = Gx / (Gx.mean(dim=-1, keepdim=True) + 1e-6)
+        return self.gamma * (inputs * Nx) + self.beta + inputs
+
+
+class GlobalResponseResidualBlock(nn.Module):
+    def __init__(self, inp_channels, channel_skip=None, kernel_size=3, dropout=0.0) -> None:
+        super().__init__()
+
+        # depthwise
+        self.depthwise = nn.Conv2d(
+            inp_channels + channel_skip,
+            inp_channels,
+            kernel_size=kernel_size,
+            padding=kernel_size // 2,
+            groups=inp_channels,
+        )
+        self.norm = nn.LayerNorm(inp_channels, elementwise_affine=False, eps=1e-6)
+
+        # channelwise
+        self.channelwise = nn.Sequential(
+            nn.Linear(inp_channels, inp_channels * 4),
+            nn.GELU(),
+            GlobalResponseNorm(inp_channels * 4),
+            nn.Dropout(dropout),
+            nn.Linear(inp_channels * 4, inp_channels),
+        )
+
+    @staticmethod
+    def _norm(x, norm):
+        return norm(x.permute(0, 2, 3, 1)).permute(0, 3, 1, 2)
+
```

Comment:
```suggestion
```

Think it's not worth adding a function for a single use case - can we just copy paste it below? :-)